mlabs/src/Mlabs/NFT/Contract/CloseAuction.hs

```diff
@@ -45,6 +46,22 @@ closeAuction symbol (AuctionCloseParams nftId) = do
   auctionState <- maybe (Contract.throwError "Can't close: no auction in progress") pure mauctionState
 
   userUtxos <- getUserUtxos
+  feeRate <- getCurrFeeRate symbol
+  (bidDependentTxConstraints, bidDependentLookupConstraints) <- do
```

Comment:
What do you think about decoupling this into a separate `Contract`? It can be part of the `when` but the `Contract` might read nicer.